src/app/SharpRaven/Utilities/RuntimeInfoHelper.cs

```diff
@@ -1,46 +1,96 @@
             // e.g: .NET Framework 4.7.2633.0, .NET Native, WebAssembly
-            var version = RuntimeInformation.FrameworkDescription;
+            var runtime = new Runtime
+            {
+                RawDescription = RuntimeInformation.FrameworkDescription
+            };
 #else
-            var mono = GetFromMono();
-            var version = mono
+            var runtime = GetFromMono();
+            runtime = runtime
                 // Environment.Version: NET Framework 4, 4.5, 4.5.1, 4.5.2 = 4.0.30319.xxxxx
                 // .NET Framework 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1 =  4.0.30319.42000
                 // Not recommended on NET45+
-                ?? $".NET Framework {Environment.Version}";
+                ?? new Runtime
+                {
+                    Name = ".NET Framework",
+                    Version = Environment.Version.ToString()
+                };
 #endif
-            return version;
+
+#if NET45PLUS_REGISTRY_VERSION // .NET Framework 4.5 and later
+            if (runtime.IsNetFx())
+            {
+                runtime.Build = Get45PlusLatestInstallationFromRegistry()?.ToString();
+            }
+#endif
+
+#if !NETFX // Non .NET Framework (i.e: netstandard, netcoreapp)
+            if (runtime.IsNetCore())
+            {
+                // RuntimeInformation.FrameworkDescription returns 4.6 on .NET Core 2.0 which is counter intuitive
+                var assembly = typeof(System.Runtime.GCSettings).GetTypeInfo().Assembly;
+                var assemblyPath = assembly.CodeBase.Split(new[] { '/', '\\' }, StringSplitOptions.RemoveEmptyEntries);
+                var netCoreAppIndex = Array.IndexOf(assemblyPath, "Microsoft.NETCore.App");
+                if (netCoreAppIndex > 0 && netCoreAppIndex < assemblyPath.Length - 2)
+                {
+                    runtime.Name = ".NET Core";
+                    runtime.Version = assemblyPath[netCoreAppIndex + 1];
+                }
+            }
+#endif
+            return runtime;
         }
 
-        private static string GetFromMono()
+        private static Runtime GetFromMono()
         {
             // The implementation of Mono to RuntimeInformation:
             // https://github.com/mono/mono/blob/90b49aa3aebb594e0409341f9dca63b74f9df52e/mcs/class/corlib/System.Runtime.InteropServices.RuntimeInformation/RuntimeInformation.cs
-            // e.g; Mono 5.10.0 (Visual Studio built mono)
-            var monoVersion = Type.GetType("Mono.Runtime", false)
-                ?.GetMethod("GetDisplayName", BindingFlags.NonPublic | BindingFlags.Static)
-                ?.Invoke(null, null) as string;
+            return Type.GetType("Mono.Runtime", false)
+                    ?.GetMethod("GetDisplayName", BindingFlags.NonPublic | BindingFlags.Static)
+                    ?.Invoke(null, null)
+                is string monoVersion
```

Comment:
The line breaking here is a bit funky. 😄  Perhaps breaking the expression up a bit would make it more readable as well?

Reply:
Ok, did reformat a bit to improve readability. 
I do believe it might be about getting used to pattern matching syntax a bit.